host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -1790,6 +1833,70 @@ esp_err_t uac_host_device_open(const uac_host_device_config_t *config, uac_host_
     return ret;
 }
 
+esp_err_t uac_host_device_open_with_vid_pid(uint16_t vid, uint16_t pid, const uac_host_device_config_t *config,
+        uint32_t timeout, uac_host_device_handle_t *uac_dev_handle)
```

Comment:
The initial idea was to make it easier for users, instead of waiting for NEW_DEV event, users simply block until the desired device is connected.

@roma-jam I agree with you that since the user can implement it externally, let's give it to the user, and let the driver focus on what it's supposed to do

Will remove the argument

Reply:
I actually like the possibility of time-out in usb_open functions. If the user only ever expects one type of device to be connected, his FreeRTOS task can be as easy as
``` c
void some_usb_task (void) {
    usb_open_with_timeout(TIMEOUT_FOREVER);
    while (true) {
        do_something_usb();
    }
}
```

The CDC driver implements it here https://github.com/espressif/esp-usb/blob/master/host/class/cdc/usb_host_cdc_acm/cdc_acm_host.c#L314

But it was written long time ago and we don't _have to_ do it the same way

Reply:
Hi @tore-espressif, thanks for your proposal, while I finally removed the `timeout` parameter to make the API more focused.

Reply:
OK, no problem :)